### DISCUSSION TITLE
C# binding code is looking to join Tree-sitter family

### DISCUSSION THREAD

**Discussion by DennySun2020** [tree-sitter/tree-sitter#2676] [Bindings]
Hello Tree-sitter contributors,

Below is the the repo for c# binding, where there are the c# binding code and the test code which shows an example for how developers can consume the tree-sitter-cpp parser through c# binding and traverse the tree in post order:

https://github.com/DennySun2020/csharp-tree-sitter

A few points that may be helpful for the discussion:
1. The c# binding code is ported from our existing project, it's been being used for 6+ months and turned out to be very robust.
2. The test code is also ported from our existing project with privacy code removed.
3. The binding code and test code have been used to parse millions lines of production code smoothly.
4. We have some partners who want to use our project to build their own product, we can give them the binding code to unblock them, but it is important for us to have the code join the Tree-sitter family officially so that they can consume and test by themselves.

Please feel free to look at the c# binding and ask me any question about it. Thank you!

**Comment by amaanq** [comment]
Hi Denny, I'd be glad to upstream this if you are okay with that! Awesome to see another language having supported bindings - and with a sizable one like C# it'd be great to have upstream. 

I believe currently the process of transferring a repository requires a member with transfer access to own the repo, so first could you perhaps transfer the repo to me? Then I'll transfer it up here and give you write access

**Comment by DennySun2020** [comment]
Hi Amaan,

Thank you for being willing to help! I'd like to transfer the repo to you, can you let me know how can I do that? It'd great if you can give me write access, I really appreciate that!
Btw I am trying to find a better way to keep touch with the org, what's your suggestion?
________________________________

**Comment by amaanq** [comment]
Hi again - it's in the repository settings somewhere, Googling "github repository transfer guide" should lead you to the Github link on doing so

I'd give you collaborator access, so basically full r/w to that repo but not the org (I don't have perms anyways for org invites)

The email in my bio/commits works well, though I have been a bit behind lately (school + being very sick for a bit with an infection...) so I might've missed something if you sent an email in the last few weeks

**Reply by DennySun2020** [reply]
Thanks for the info! The c# binding code is now going through some internal legal process before releasing to open-source community, I'll let you know once it is ready for being transferred to you.

**Reply by DennySun2100** [reply]
Hi @amaanq, the c# binding code can be transferred to you now, which account of yours should i use?

**Reply by DennySun2020** [reply]
@amaanq , I have transferred the repo to your account amaanq, looks like you need to accept that on your end?

**Reply by amaanq** [reply]
> @amaanq , I have transferred the repo to your account amaanq, looks like you need to accept that on your end?

yep sorry, let me get to that

should be done!

**Reply by DennySun2020** [reply]
Thanks @amaanq, btw once you upload the repo to tree-sitter org, how about updating this page https://tree-sitter.github.io/tree-sitter/ to add c# binding there so that tree-sitter developers can know this support.

**Reply by DennySun2020** [reply]
@amaanq , could you help merge this since I have no write access - I updated the doc in this PR https://github.com/tree-sitter/tree-sitter/pull/2716

**Reply by amaanq** [reply]
@DennySun2020 are you on a platform like Matrix where we can chat? I'd like to improve the C# bindings a bit to be in-line w/ the other ones (I try to do that w/ Python and Node as well)

**Reply by DennySun2020** [reply]
@amaanq, I am not on that platform, but I can create one since we might need to interact more on tree-sitter PRs in future.

**Reply by DennySun2020** [reply]
Hi @amaanq, I sent you a message on Element, could you send a ACK
